### PR TITLE
Feature/acf 5.8 integration

### LINKED ISSUE #472: Remove Animate.css
I think we've discussed this before, but I think we should remove `animate.css` and the associated "animation" field from WD_s for a couple of reasons.
- We load all of animate.css all the time into the stylesheet.
- Animate.css only loads on initial load so blocks that utilize the animation "below the fold" will never animate.
- I don't think this feature is utilized with any great amount.

If we decide to keep it I think we should:
- Work in some javascript to trigger the animation on block display into view.
- Only enqueue what we need to when we need to (somehow) instead of the entire animate.css stylesheet.

What say we?

### PR DESCRIPTION
**Pull request by coreymcollins**
Closes #472 (sneaky sneaky)

### DESCRIPTION ###
- This moves our ACF Flexible Content Blocks out of post meta and into the post as Gutenberg blocks. 
- Removes some of our ACF block options (like setting a class or an ID, as this can be done with core Gutenberg fields)
- Restricts our fields to two layouts – wide and full. Wide is default and fits to the content container. Full expands to the full width of the browser.
- Sets the blocks up to have them always visibly styled/laid out in the backend with field editing happening in the sidebar by default. You can always click the pencil icon to edit the block in its place in the post.
- Converted our tabs to accordions for better sidebar layout and overall better  cleanliness in the Gutenberg layout
- **Removes Animate.css!** We barely ever use this and it's kind of become silly to include. If we want to add animate-on-scroll functionality, we could work toward that. But, a one-time animation on page load is not super useful or used all that often in my experience.
- This adds an okay-for-now solution to adding our frontend styles to our blocks on the backend. Since we have some open tickets and PRs for moving away from Gulp, I didn't want to go too deep into the Gulp Hole in trying to come up with a crazy solution. Open to suggestion, of course!

### SCREENSHOTS ###

An active block in the backend:

Or, you can edit it in its place:

### OTHER ###
- [x] Is this issue accessible? (Section 508/WCAG 2.0AA)
- [x] Does this issue pass all the linting? (PHPCS, ESLint, SassLint)
- [x] Does this pass CBT? (https://app.crossbrowsertesting.com/public/i29947718af4369d/screenshots/z2352a850832947fb698)

### STEPS TO VERIFY ###
- Checkout the branch
- Create a page or post with Gutenberg blocks (our blocks are all in a WDS Blocks category)
- Test background options, start/date time options, etc.
- Kick the crap out of it!

### DOCUMENTATION ###
Will this pull request require updating the wd_s [wiki](https://github.com/WebDevStudios/wd_s/wiki)?

Yes, for sure. I can start prepping some base level documentation updates, but won't get too in the weeds in the event that something changes drastically.

### CODE REVIEW

**Comment by gregrickaby** [comment]
screenshot

File: inc/acf-gutenberg.php [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+	$supports = array(
+		'align'  => array( 'wide', 'full' ),
+		'anchor' => true,
+	);
+
+	// Register our Accordion block.
+	acf_register_block_type(
+		array(
+			'name'            => 'wds-accordion',
+			'title'           => __( 'Accordion', '_s' ),
```
Comment (gregrickaby) [inline inc/acf-gutenberg.php]: Please use `esc_html__()` instead of `__()` ... throughout the whole file.

File: inc/acf-gutenberg.php [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+ * @return void Bail if the block has expired.
+ * @author Corey Collins
+ */
+function _s_acf_block_registration_callback( $block ) {
+
+	// Convert the block name into a handy slug.
+	$block_slug = str_replace( 'acf/', '', $block['name'] );
+
+	// If the block has expired,then bail!
+	if ( _s_has_block_expired(
```
Comment (gregrickaby) [inline inc/acf-gutenberg.php]: When a block is scheduled in the future (or has expired), it no longer shows up in the block editor, until you make a change. I think the block should appear in the block editor, no matter what.

screenshot

**Review by gregrickaby — COMMENTED** [review]
Initial pass is excellent. Great work, @coreymcollins!

File: inc/acf-gutenberg.php (hunk 2 above)
Comment (coreymcollins) [reply inc/acf-gutenberg.php]: Good call. Even better hero.

**Comment by gregrickaby** [comment]
Because the page data is now saved inside `post_content` vs. `post_meta`, do we still need this? https://github.com/WebDevStudios/wd_s/blob/c3e7ee0c2771b9d7c06dd87b433aa733c7d5ff01/inc/acf-search.php#L19

I realize that it's not really hurting anything...

**Comment by coreymcollins** [comment]
> Because the page data is now saved inside `post_content` vs. `post_meta`, do we still need this?
> 
> https://github.com/WebDevStudios/wd_s/blob/c3e7ee0c2771b9d7c06dd87b433aa733c7d5ff01/inc/acf-search.php#L19
> 
> I realize that it's not really hurting anything...

I thought about removing that, but decided to keep it in the event that we utilize actual post meta via ACF that needs to be found via search. We can always toss it if we decide that's not a need.

**Comment by coreymcollins** [comment]
Pushed some updates which should resolve the two other issues noted above.

I also added some messaging on the backend when a block is expired:


The messaging _does_ go away as soon as you remove the dates, but it doesn't appear again until you update the post and refresh after adding dates. I want to dig in and see if I can find out if there's a way to check for those updates in real time to add the message without needing to save and reload.

**Comment by coreymcollins** [comment]
[This commit](https://github.com/WebDevStudios/wd_s/pull/480/commits/07742a8c6d063faa1dc0be92270c3357e3f234be) adds the ability to update the expired messaging on the fly. Instead of looking for block data, it looks for the ACF field which updates on the fly rather than on post save/page reload.

In action:

**Comment by ImBigWill** [comment]
From what I've tested so far, things look great. I think one thing to consider on the backend may be to add a little space between the blocks if they're not in preview mode (maybe even if they are) to improve the user experience some?

Screen Shot 2019-08-02 at 4 52 38 PM
I will continue looking into this further but just had that one note for now.

**Comment by coreymcollins** [comment]
Just pushed an update to resolve Will's note above. Gutenberg itself doesn't add/change any classes when switching to Edit mode, but I found a way to target ACF classes to give these blocks some breathing room when in Edit mode.

**Comment by gregrickaby** [comment]
@coreymcollins Everything is looking good on my end. I've also spoken with Brad/Jodie on the sales front, and they're aware of this PR and the implications. They were both _very_ excited about being Gutenberg ready!

Great work! 💪

**Comment by coreymcollins** [comment]
In addition to the spacing for the Edit view, I've added an extra field to each block. Using the "Messages" field, I'm adding the name of each block type so you can see which block is which while editing:




The latest commits also add default values to each of the blocks so they don't appear empty when first being added:

**Review by gregrickaby — APPROVED** [review]

**Review by ImBigWill — APPROVED** [review]

**Comment by coreymcollins** [comment]
Thanks for the reviews @gregrickaby & @ImBigWill – gonna get this puppy merged in this afternoon!

💥

**Comment by coreymcollins** [comment]
Merged in manually to resolve compiled file conflicts when merging into master.

Closing out!

**Pull request closed without merging**